docs/devel/scheduler.md

```diff
@@ -41,45 +41,67 @@ indicating where the Pod should be scheduled.
 
 ## The scheduling process
 
-The scheduler tries to find a node for each Pod, one at a time, as it notices
-these Pods via watch. There are three steps. First it applies a set of "predicates" that filter out
-inappropriate nodes. For example, if the PodSpec specifies resource requests, then the scheduler
-will filter out nodes that don't have at least that much resources available (computed
-as the capacity of the node minus the sum of the resource requests of the containers that
-are already running on the node). Second, it applies a set of "priority functions"
-that rank the nodes that weren't filtered out by the predicate check. For example,
-it tries to spread Pods across nodes and zones while at the same time favoring the least-loaded
-nodes (where "load" here is sum of the resource requests of the containers running on the node,
+```
+                          +-------+
+          +---------------+ node 1|
+          |               +-------+
+          |
+   +----> |  Apply pred. filters
+   |      |
+   |      |                +-------+
+   |      +----+---------->+node 2 |
+   |           |           +--+----+
+   |  watch    |              |
+   |           |              |        +------+
+   |           +---------------------->+node 3|
++--+----------+               |        +--+---+
+| Pods in etcd|               |           |
```

Comment:
change this to Pods in API sever? We want to hide etcd for components outside API server for most the times.